frontend/server/server.ts

```diff
@@ -70,24 +79,37 @@ const MINIO_ENDPOINT = MINIO_NAMESPACE && MINIO_NAMESPACE.length > 0 ? `${MINIO_
 const _as_bool = (value: string) => ['true', '1'].indexOf(value.toLowerCase()) >= 0
 
 /** minio client for minio storage */
-const minioClient = new MinioClient({
+const minioOptions: MinioClientOptions = {
   accessKey: MINIO_ACCESS_KEY,
   endPoint: MINIO_ENDPOINT,
   port: parseInt(MINIO_PORT, 10),
   secretKey: MINIO_SECRET_KEY,
   useSSL: _as_bool(MINIO_SSL),
-} as any);
+};
+const minioClient = new MinioClient(minioOptions);
 
 /** minio client for s3 objects */
-const s3Client = new MinioClient({
+const s3Options: MinioClientOptions = {
   endPoint: 's3.amazonaws.com',
   accessKey: AWS_ACCESS_KEY_ID,
   secretKey: AWS_SECRET_ACCESS_KEY,
-} as any);
+};
+/** minio client to s3 with AWS instance profile IAM if access keys are not provided. */
+const s3ClientPromise = () => createMinioClient(s3Options);
```

Comment:
changed this to a callable to `Promise<MinioClient>`. So that the credentials will be correctly updated if the token had expired.

Reply:
I don't know enough background. Is minio client completely stateless?
For example, when there are two requests coming, each request uses a different minio client. Could this be a problem?

Feel free to go ahead if it's not an issue

Reply:
Yup. It will not be an issue. 

The client actually just constructs the authentication header as well as the corresponding http request based on the configs.

Reply:
Cool. Sounds good to me